dashboard/app/models/course_offering.rb

```diff
@@ -60,4 +66,42 @@ def self.get_from_cache(key)
       CourseOffering.find_by_key(key)
     end
   end
+
+  def serialize
+    {
+      key: key,
+      display_name: display_name
+    }.merge(properties&.sort.to_h)
+  end
+
+  def write_serialization
+    return unless Rails.application.config.levelbuilder_mode
+    file_path = Rails.root.join("config/course_offerings/#{key}.json")
+    object_to_serialize = serialize
+    dirname = File.dirname(file_path)
+    unless File.directory?(dirname)
+      FileUtils.mkdir_p(dirname)
+    end
+    File.write(file_path, JSON.pretty_generate(object_to_serialize))
+  end
+
+  def self.seed_all(glob="config/course_offerings/*.json")
+    removed_records = all.pluck(:key)
+    Dir.glob(Rails.root.join(glob)).each do |path|
+      removed_records -= [CourseOffering.seed_record(path)]
+    end
+    where(key: removed_records).destroy_all
+  end
+
+  def self.properties_from_file(content)
+    config = JSON.parse(content)
+    config.symbolize_keys
+  end
+
+  def self.seed_record(file_path)
+    properties = properties_from_file(File.read(file_path))
+    course_offering = CourseOffering.find_or_initialize_by(key: properties[:key])
+    course_offering.update! properties
+    course_offering.key
+  end
```

Comment:
please note in method level comment that you are returning the key, since that's a bit unexpected for a method which just says it is seeding